### PR TITLE
chore(logwatch): Add `ignore.conf` to ignore logs from Dovecot `index-worker`

### PR DESCRIPTION
**Pull request by mmehnert**
When using the dovecot-fts-xapian plugin as described in the "Full-Text Search" article of the projects' documentation, the daily logwatch emails get cluttered with output from the indexing of emails. 
This pull request adds an ignore.conf to the logwatch configuration with a regular expression to filter out the corresponding messages. 

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] **I have added information about changes made in this PR to `CHANGELOG.md`**

### CODE REVIEW

**Comment by georglauterbach** [comment]
Thank you for the PR, this looks like a sensible change.

However, the checklist is not supposed to be shortened, **it's supposed to be completed!** I have re-added the missing points, please make sure to complete all the items and **check** them in the list. I have already ticked the items in the list that can be ticked.

File: target/logwatch/ignore.conf
Comment (georglauterbach) [inline target/logwatch/ignore.conf:1]: Please add a brief comment explaining that this has to do with Xapian.

**Review by georglauterbach — CHANGES_REQUESTED** [review]
LGTM 👍🏼 One small change and we're good I think.

**Comment by mmehnert** [comment]
> Thank you for the PR, this looks like a sensible change.
> 
> However, the checklist is not supposed to be shortened, **it's supposed to be completed!** I have re-added the missing points, please make sure to complete all the items and **check** them in the list. I have already ticked the items in the list that can be ticked.

I'm sorry that I mishandled that. I can't see the missing points / check them in the list. Should I copy them from somewhere again and modify the pull request text myself?

**Comment by polarathene** [comment]
> I can't see the missing points / check them in the list. Should I copy them from somewhere again and modify the pull request text myself?

I've restored these.

I'm not sure if @georglauterbach is wanting them all addressed though (tests/docs), they don't seem necessary.

You should add a changelog entry, this seems suitable for the "Updates" section:

https://github.com/docker-mailserver/docker-mailserver/blob/016d6b5255be6945b08616a79f92f496a583b775/CHANGELOG.md?plain=1#L84-L85

Just add something above the referenced like, such as:

```md
- **Dovecot:**
  - `logwatch` now filters out non-error logs related to the status of the `index-worker` process for FTS indexing.
```

---

Regarding this contribution, do you have some samples of the logs you're preventing? I have [seen here](https://github.com/grosjo/fts-xapian/issues/8#issue-401208380) that there can be error logs that may be relevant to not ignore if they appear?

```
Jan 21 06:35:24 mailserver dovecot: indexer-worker: Error: terminate called after throwing an instance of 'std::bad_alloc'
Jan 21 06:35:24 mailserver dovecot: indexer-worker: Error: what(): std::bad_alloc
Jan 21 06:35:24 mailserver dovecot: indexer: Error: Indexer worker disconnected, discarding 1 requests for email@example.com
Jan 21 06:35:24 mailserver dovecot: indexer-worker(email@example.com)<16330><rgw7KfF/AJsqCqVDQq0AAD1inuN9Ly/l:UJw9Kz5ZRVzKPwAAhwicDQ>: Fatal: master: service(indexer-worker): child 16330 killed with signal 6 (core dumps disabled - https://dovecot.org/bugreport.html#coredumps)
Jan 21 06:35:24 mailserver dovecot: imap(email@example.com)<16326><rgw7KfF/AJsqCqVDQq0AAD1inuN9Ly/l>: Error: indexer failed to index mailbox INBOX
```

Whereas the goal seems to be more targeted around filtering these kind of logs out:

```
Jan 21 06:31:34 mailserver dovecot: indexer-worker(email@example.com)<16330><ZN42KfF//poqCqVDQq0AAD1inuN9Ly/l:Wut/Mf9YRVzKPwAAhwicDQ>: Indexed 5969 messages in Gesendet (UIDs 1..5972)
Jan 21 06:31:34 mailserver dovecot: indexer-worker(email@example.com)<16330><ZN42KfF//poqCqVDQq0AAD1inuN9Ly/l:8BYpLzZZRVzKPwAAhwicDQ>: Start indexing 'Papierkorb' (/srv/mailstore/example.com/email/xapian-indexes/db_90dda8344dfc3158af39000087089c0d)
Jan 21 06:31:42 mailserver dovecot: indexer-worker(email@example.com)<16330><ZN42KfF//poqCqVDQq0AAD1inuN9Ly/l:8BYpLzZZRVzKPwAAhwicDQ>: Done indexing 'Papierkorb'
Jan 21 06:31:42 mailserver dovecot: indexer-worker(email@example.com)<16330><ZN42KfF//poqCqVDQq0AAD1inuN9Ly/l:8BYpLzZZRVzKPwAAhwicDQ>: Indexed 1012 messages in Papierkorb (UIDs 1..3588)
Jan 21 06:31:42 mailserver dovecot: indexer-worker(email@example.com)<16330><rgw7KfF/AJsqCqVDQq0AAD1inuN9Ly/l:UJw9Kz5ZRVzKPwAAhwicDQ>: Start indexing 'INBOX' (/srv/mailstore/example.com/email/xapian-indexes/db_28c518021bee30580b02000087089c0d)
```

Keywords in these logs are `dovecot: indexer-worker` + `Indexed` / `Start indexing` (_or `xapian-indexes`, possibly too specific_) / `Done indexing`.

I see the regex is targeting one keyword pair there but will still include the start/done log lines, is that still desired? The error logs should remain, so no problem there.

---

For maintainers - Reference links related to the Dovecot indexer process and FTS:
- https://doc.dovecot.org/configuration_manual/service_configuration/#indexer
- https://doc.dovecot.org/3.0/man/doveadm-indexer.1/
- https://doc.dovecot.org/3.0/man/doveadm-index.1/
- https://doc.dovecot.org/configuration_manual/fts/

**Comment by georglauterbach** [comment]
Indeed, we don't no l need most of the points, so I ticked them, too. The `CHANGELOG` update is the most important one :)

**Comment by polarathene** [comment]
> so I ticked them, too

That'd be misleading as no such changes were done. Removing those lines is fine, I've edited to drop those redundant checklist items.

@mmehnert just update `CHANGELOG.md` as per suggestion above and consider the feedback on the regex, you should be good to go.

**Comment by mmehnert** [comment]
> Regarding this contribution, do you have some samples of the logs you're preventing? I have [seen here](https://github.com/grosjo/fts-xapian/issues/8#issue-401208380) that there can be error logs that may be relevant to not ignore if they appear?

I was explicitly only targeting messages of the form 

`Jan 21 06:31:34 mailserver dovecot: indexer-worker(email@example.com)<16330><ZN42KfF//poqCqVDQq0AAD1inuN9Ly/l:Wut/Mf9YRVzKPwAAhwicDQ>: Indexed 5969 messages in Gesendet (UIDs 1..5972)`

Interestingly in my setup I don't have the messages containing "Start indexing" and "Done indexing".  So I did not include a regexp for those.  @polarathene do you encounter those messages in your installation? If so, I'm happy to add the necessary two lines to ignore.conf

The error messages like 

`Jan 21 06:35:24 mailserver dovecot: indexer-worker: Error: terminate called after throwing an instance of 'std::bad_alloc'`

should not be excluded in my opinion and are not matched by the regexp that I've put into ignore.conf

**Review by polarathene — APPROVED** [review]

**Comment by polarathene** [comment]
Thanks for the contribution! 😁 

I'll leave this for another maintainer to approve before merging.

> @polarathene do you encounter those messages in your installation?

I don't actually deploy DMS (ironic I know), and I can't configure a local instance with FTS to try it out myself any time soon 😓 

If those log lines aren't present, I wouldn't worry about it. Either something changed since that report or we have a less verbose log from Dovecot configured. If someone is affected and it bothers them they're welcome to contribute another PR 👍

**Review by georglauterbach — APPROVED** [review]

**Pull request merged**
